python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -275,6 +275,8 @@ def get_target_root_partitions_subset(
             asset_key
             for asset_key in self.target_subset.asset_keys
             if asset_graph.get_partitions_def(asset_key) is not None
+            # Check that the asset was partitioned at backfill creation time
+            and asset_key in self.target_subset.partitions_subsets_by_asset_key
```

Comment:
Do we even need the `asset_graph.get_partitions_def()` check here? I.e. if something went from partitioned -> unpartitioned (so the historical data is partitioned, but the current asset graph has no partitions defs), would we still be able to show historical data?

mostly curious

Reply:
Yeah, we are able to show historical data now that we store time partitions defs on the subset!

I'll rework this based on your suggestion, thanks